src/txmempool.h

```diff
@@ -511,15 +511,6 @@ class CTxMemPool
      * `mempool.cs` whenever adding transactions to the mempool and whenever
      * changing the chain tip. It's necessary to keep both mutexes locked until
      * the mempool is consistent with the new chain tip and fully populated.
-     *
-     * @par Consistency bug
-     *
-     * The second guarantee above is not currently enforced, but
-     * https://github.com/bitcoin/bitcoin/pull/14193 will fix it. No known code
-     * in bitcoin currently depends on second guarantee, but it is important to
-     * fix for third party code that needs be able to frequently poll the
```

Comment:
In commit "validation: Add missing mempool locks" (fadeeadd93a928c8ececa8c96ad2afa45bd25a29)

Might be worth keeping text that says why guarantee 2 is useful. James also asked about it here https://github.com/bitcoin/bitcoin/pull/14193#pullrequestreview-179871886, wondering if it might be useful for dandelion.

Reply:
Not sure about that. We generally don't update (forget to update) comments when adding new features. So if a feature is added in the far future (e.g. Dandelion has no chance to hit any time soon), the phrase "No known code in bitcoin currently depends on second guarantee" might no longer reflect reality.

I believe the test that I added serves a a nice documentation that if this guarantee is violated, it will fail. The motivation to fix this bug for (enterprise) rpc users that poll the mempool should be enough motivation on its own.